datastore/src/main/java/io/spine/server/storage/datastore/DsFilters.java

```diff
@@ -169,20 +168,42 @@ static Filter activeEntity() {
      */
     private static Collection<Filter> toFilters(Collection<CompositeQueryParameter> parameters,
                                                 ColumnFilterAdapter columnFilterAdapter) {
-        FluentIterable<CompositeQueryParameter> params = from(parameters);
-        FluentIterable<CompositeQueryParameter> conjunctionParams =
-                params.filter(isConjunctive);
-        FluentIterable<CompositeQueryParameter> disjunctionParams =
-                params.filter(isDisjunctive);
-        Optional<CompositeQueryParameter> firstParam = conjunctionParams.first();
+        List<CompositeQueryParameter> conjunctionParams = parameters.stream()
+                                                                    .filter(isConjunctive)
+                                                                    .collect(toList());
+        List<CompositeQueryParameter> disjunctionParams = parameters.stream()
+                                                                    .filter(isDisjunctive)
+                                                                    .collect(toList());
         Optional<CompositeQueryParameter> mergedConjunctiveParams =
-                firstParam.transform(new ParameterReducer(conjunctionParams.skip(1)));
+                mergeConjunctiveParameters(conjunctionParams);
+
         Collection<Filter> filters = newLinkedList();
         TreePathWalker processor = new ColumnFilterReducer(columnFilterAdapter, filters);
-        multiply(mergedConjunctiveParams.orNull(), disjunctionParams, processor);
+        multiply(mergedConjunctiveParams.orElse(null), disjunctionParams, processor);
         return filters;
     }
 
+    /**
+     * Merges conjunctive parameters into a single {@link Filter}.
+     *
+     * @param conjunctiveParams list of parameters
+     * @return resulting filter or {@code Optional.empty()} if there {@code conjunctiveParams} is
+     * empty
+     */
+    private static Optional<CompositeQueryParameter> mergeConjunctiveParameters(
```

Comment:
I don't understand why this is in `gcloud-java`, while this should most likely be in `core-java`.

Let's discuss in person.

Reply:
As discussed, this should be reflected as an issue. LGTM for now.